wagtail/permission_policies/collections.py

```diff
@@ -9,6 +9,28 @@
 
 
 class CollectionPermissionLookupMixin:
+    permission_cache_name = "_collection_permission_cache"
+
+    def _get_user_permission_objects_for_actions(self, user, actions):
+        """
+        Get a set of the user's GroupCollectionPermission objects for the given actions
+        """
+        permission_codenames = {
+            get_permission_codename(action, self.auth_model._meta) for action in actions
+        }
+        return {
+            group_permission
+            for group_permission in self.get_cached_permissions_for_user(user)
+            if group_permission.permission.codename in permission_codenames
+        }
+
+    def get_all_permissions_for_user(self, user):
+        if not user.is_active or user.is_anonymous or user.is_superuser:
+            return GroupCollectionPermission.objects.none()
```

Comment:
question: wouldn't superusers have all permissions?

Reply:
Yes, but they don't necessarily need to have a associated `GroupCollectionPermission` (as they don't even need groups).

The permission check itself happens in `_check_perm()`, and we short-circuit it for these users

Reply:
Added